presto-session-property-managers/src/main/java/io/prestosql/plugin/session/AbstractSessionPropertyManager.java

```diff
@@ -49,7 +50,22 @@
     @Override
     public Map<String, Map<String, String>> getCatalogSessionProperties(SessionConfigurationContext context)
     {
-        // NOT IMPLEMENTED YET
-        return ImmutableMap.of();
+        Map<String, Map<String, String>> catalogsSessionProperties = new HashMap<>();
```

Comment:
Use ImmutableMap.Builder

Reply:
Partially done
Tried using groupingby collector or a custom collector to do it in a single path but only came up with more verbose & harder to grasp implementations.